src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutes.scala

```diff
@@ -339,34 +338,28 @@ trait ResourceRoutes extends UserInfoDirectives with SecurityDirectives with Sam
   def setResourceParent(resource: FullyQualifiedResourceId, userInfo: UserInfo, samRequestContext: SamRequestContext): server.Route =
     put {
       entity(as[FullyQualifiedResourceId]) { newResourceParent =>
-        withCurrentParentOption(resource, samRequestContext) {
-          case Some(currentResourceParent) =>
-            requireAction(currentResourceParent, SamResourceActions.removeChild, userInfo.userId, samRequestContext) {
-              setResourceParentInternal(resource, newResourceParent, userInfo, samRequestContext)
+        requireAction(resource, SamResourceActions.setParent, userInfo.userId, samRequestContext) {
+          requireParentAction(resource, None, SamResourceActions.removeChild, userInfo.userId, samRequestContext) {
+            requireParentAction(resource, Option(newResourceParent), SamResourceActions.addChild, userInfo.userId, samRequestContext) {
+              complete(resourceService.setResourceParent(resource, newResourceParent, samRequestContext).map(_ => StatusCodes.NoContent))
             }
-          case None => setResourceParentInternal(resource, newResourceParent, userInfo, samRequestContext)
+          }
         }
       }
     }
 
-  private def setResourceParentInternal(resource: FullyQualifiedResourceId, newResourceParent: FullyQualifiedResourceId, userInfo: UserInfo, samRequestContext: SamRequestContext): server.Route = {
-    requireAction(newResourceParent, SamResourceActions.addChild, userInfo.userId, samRequestContext) {
-      requireAction(resource, SamResourceActions.setParent, userInfo.userId, samRequestContext) {
-        complete(resourceService.setResourceParent(resource, newResourceParent, samRequestContext).map(_ => StatusCodes.NoContent))
-      }
-    }
-  }
-
   def deleteResourceParent(resource: FullyQualifiedResourceId, userInfo: UserInfo, samRequestContext: SamRequestContext): server.Route =
     delete {
-      withCurrentParentOption(resource, samRequestContext) {
-        case Some(resourceParent) =>
-          requireAction(resourceParent, SamResourceActions.removeChild, userInfo.userId, samRequestContext) {
-            requireAction(resource, SamResourceActions.setParent, userInfo.userId, samRequestContext) {
-              complete(resourceService.deleteResourceParent(resource, samRequestContext).map(_ => StatusCodes.NoContent))
+      requireAction(resource, SamResourceActions.setParent, userInfo.userId, samRequestContext) {
+        requireParentAction(resource, None, SamResourceActions.removeChild, userInfo.userId, samRequestContext) {
```

Comment:
I think that `requireParentAction(resource, None` will always pass. Instead of `None` I think we need to to pass in a `currentParent: Option[FullyQualifiedResourceId]` or look for it within `requireParentAction`

Reply:
see SecurityDirectives line 93: when None is passed it will look up the current parent

Reply:
I think for `setResourceParent`, the resulting error would be: `StatusCodes.BadRequest, "Cannot set parent as this would introduce a cyclical resource hierarchy"`.

Reply:
(I missed the parameter name `newParent` in `requireParentAction`)